Octokit/Http/Connection.cs

```diff
@@ -136,6 +136,30 @@ public Connection(ProductHeaderValue productInformation, Uri baseAddress, ICrede
             _jsonPipeline = new JsonHttpPipeline();
         }
 
+        /// <summary>
+        /// Gets the latest API Info - this will be null if no API calls have been made
+        /// </summary>
+        /// <returns><seealso cref="ApiInfo"/> representing the information returned as part of an Api call</returns>
+        public ApiInfo LastApiInfo
+        { 
+            get
+            {
+                lock (LastApiInfoLocker)
+                {
+                    return _lastApiInfo;
+                }
+            }
+            private set
+            {
+                lock (LastApiInfoLocker)
+                {
+                    _lastApiInfo = value;
+                }
+            }
+        }
+        private ApiInfo _lastApiInfo;
+        private readonly object LastApiInfoLocker = new object();
+
```

Comment:
:thumbsup:

Let's remove the locks, but make sure we comment it and explain why so if someone else comes along, they understand why there's shared mutable state.

Also, one idea we should do is to set `LastApiInfo` to a copy of the `ApiInfo` to reduce the sharing of mutable state. In fact, I think `LastApiInfo` should be a method `GetLastApiInfo` and it should always return a copy of the last `ApiInfo`

That way, Octokit is in control of the lifetime of that object and a consumer can't inadvertently keep the object around longer than expected. Thoughts?